### PR TITLE
Windows modular observ lib

### PR DESCRIPTION
**Pull request by v-zhuravlev**
`cd windows-mixin`
`grr apply mixin.libsonnet`

Checklist:
- [x] Fleet dashboard
  - [x] Add drill down and data links to other dashes
- [x] Overview dashboard
- [x] Networking dashboard (WONT DO, don't have many network metrics)
- [x] Disks dashboard
- [x] Memory dash (WONT DO, requires additional collector)
- [x] cpu and system dash
  - [x] NTP and time metrics (enable `time` collector)
- [x] Logs dashboard
  - [x] Hide under feature flag (EnableLokiLogs)
- [ ] Alerts:
  - [x] Linux parity: Time alerts (no ntp, offset too big) 
  - [x] Physical disk status (enable disk module) (enable disk module)
  - [ ] Linux parity: add disk predict alert
- [x] Extras:
  - [x] Annotations: Reboot,failed service, critical event
  - [x] Logs scrape snippet
  - [x] Add descriptions to panels

Changelog:

- Create three-tier dashboards (Fleet->Overview->Advanced drill down, including logs), similar to node exporter dashboards
- Introduced modular observ lib structure, that should make this lib highly customizable and reusable
- Increase pending period for alerts to 15m to match linu
- Add disk status alert
- Add common annotations: reboot,failedService,criticalSystemEvent

Examples:
Fleet dash:
image
Overview dash:
image
Logs dash:
image
Disks dash:
image

### CODE REVIEW

File: windows-mixin/jsonnetfile.json
Comment (rgeyer) [inline windows-mixin/jsonnetfile.json:11]: You'll want to have a follow-up PR to change this to master, or perhaps you can reference the specific commit SHA. I believe the branch usually gets deleted on merge.

File: windows-observ-lib/alerts.libsonnet
Comment (rgeyer) [inline windows-observ-lib/alerts.libsonnet:4]: Looks like these are all done, this can be removed?
```suggestion
```

File: windows-observ-lib/dashboards.libsonnet
Comment (rgeyer) [inline windows-observ-lib/dashboards.libsonnet:82]: Is there a problem with these? Should they also be removed from the windows-observ-lib?

File: windows-observ-lib/jsonnetfile.json
Comment (rgeyer) [inline windows-observ-lib/jsonnetfile.json:11]: Should this get merged first? Given it's a requirement, we should probably have that nailed down with some certainty that the interface will not change.

File: windows-observ-lib/panels.libsonnet
Comment (rgeyer) [inline windows-observ-lib/panels.libsonnet:193]: Same question as in `dashboards.libsonnet`.. If these are unused, should they persist?

File: windows-observ-lib/panels.libsonnet
Comment (rgeyer) [inline windows-observ-lib/panels.libsonnet:233]: Seems like the description adequately explains why this metric is important?
```suggestion
```

File: windows-observ-lib/utils.libsonnet
Comment (rgeyer) [inline windows-observ-lib/utils.libsonnet:1]: I think most of these are pretty common utils. I don't recall. Did we intend to merge these with grafonnet or with the common-lib?

**Review by rgeyer — COMMENTED** [review]
Looks good overall. A few nits here and there.

Mostly, I think we should get the dependencies reviewed/merged before this is merged. What do you think?

File: windows-observ-lib/alerts.libsonnet
Comment (v-zhuravlev) [inline windows-observ-lib/alerts.libsonnet:136]:
```suggestion
                'NTP client on instance {{ $labels.instance }} is not running.'
```

File: windows-observ-lib/alerts.libsonnet
Comment (v-zhuravlev) [inline windows-observ-lib/alerts.libsonnet:104]:
```suggestion
                'Round-trip time of NTP client on instance {{ $labels.instance }} is greater than 1 second. Delay is {{ $value }} sec.'
```

File: windows-observ-lib/alerts.libsonnet
Comment (v-zhuravlev) [inline windows-observ-lib/alerts.libsonnet:120]:
```suggestion
              summary: 'NTP time offset is too large'
              description: |||
                'NTP time offset for instance {{ $labels.instance }} is greater than 1 second. Offset is {{ $value }} sec.'
```

File: windows-observ-lib/jsonnetfile.json
Comment (v-zhuravlev) [reply windows-observ-lib/jsonnetfile.json:11]: commonlib is ready for review. once we merge it, we can continue with windows lib.

File: windows-observ-lib/panels.libsonnet
Comment (v-zhuravlev) [reply windows-observ-lib/panels.libsonnet]: thnks, removed

File: windows-observ-lib/utils.libsonnet
Comment (v-zhuravlev) [reply windows-observ-lib/utils.libsonnet]: Thnks, Moved to commonlib

**Review by rgeyer — APPROVED** [review]
This looks ready to go!

I'd wait until common-lib is merged, and update that reference in jsonnet bundler, but otherwise. 🚀

File: windows-observ-lib/README.md
Comment (v-zhuravlev) [reply windows-observ-lib/README.md:98]: for integration, yes.

File: windows-observ-lib/dashboards.libsonnet
Comment (v-zhuravlev) [reply windows-observ-lib/dashboards.libsonnet:80]: I'll check and make follow-up PR.

**Pull request merged**
